core/src/Streamly/Internal/FileSystem/Posix/ReadDir.hsc

```diff
@@ -59,9 +59,13 @@ data {-# CTYPE "struct dirent" #-} CDirent
 
 newtype DirStream = DirStream (Ptr CDir)
 
+-------------------------------------------------------------------------------
+-- readdir operations
 -------------------------------------------------------------------------------
 
-foreign import ccall unsafe "closedir"
+-- IMPORTANT NOTE: Use capi FFI for all readdir APIs. This is required at
```

Comment:
You can explain why a little more here.

Reply:
that may make the note too long and specific, one can read the difference between capi and ccall in GHC manual.